src/firefly/js/visualize/ui/ColorDialog.jsx

```diff
@@ -2,248 +2,226 @@
-        var range= (zscale.value==='zscale') ? this.renderZscale() : this.getUpperAndLowerFields();
-        return (
-            <div>
-                {range}
-                <div style={{paddingTop:'10px'}}/>
-                <ValidationField  labelWidth={LABEL_WIDTH} fieldKey='gamma' groupKey={groupKey} />
-            </div>
-        );
-    },
-
-    renderZscale() {
-        var {groupKey}=this.props;
-        return (
-            <div>
-                <ValidationField  labelWidth={LABEL_WIDTH} fieldKey='contrast' groupKey={groupKey} />
-                <ValidationField  labelWidth={LABEL_WIDTH} fieldKey='numSamp' groupKey={groupKey} />
-                <ValidationField  labelWidth={LABEL_WIDTH} fieldKey='sampPerLine' groupKey={groupKey} />
-            </div>
-        );
-    },
+function makeSerializedRv(request) {
+    const useZ= Boolean(request.zscale);
 
-    render() {
-        var groupKey=this.props.groupKey;
-        var fields= FieldGroupUtils.getGroupFields(groupKey);
-        var panel;
-        if (fields) {
-            const {algorithm, lowerType, zscale}=fields;
-            var a= Number.parseInt(algorithm.value);
-            if (a===STRETCH_ASINH) {
-                panel= this.renderAsinH();
-            }
-            else if (a===STRETCH_POWERLAW_GAMMA) {
-                panel= this.renderGamma();
-            }
-            else if (zscale.value==='zscale') {
-                panel= this.renderZscale();
-            }
-            else {
-                panel= this.renderStandard();
-            }
-        }
-        else {
-            panel= this.renderStandard();
-        }
-        return (
-            <FieldGroup groupKey={groupKey} reducerFunc={colorPanelChange(this.props.band)}
-                        validatorFunc={null} keepState={true}>
-                <div style={{padding:'5px'}}>
-                    <div style={{display:'table', margin:'auto auto'}}>
-                        {this.getStretchTypeField()}
-                    </div>
-                    {panel}
-                    <div style={{display:'table', margin:'auto auto', paddingTop:'20px'}}>
-                        <CheckboxGroupInputField
-                            options={ [ {label: 'Use ZScale for bounds', value: 'zscale'} ] }
-                            fieldKey='zscale'
-                            labelWidth={0}
-                            groupKey={groupKey}/>
-                    </div>
-                </div>
-            </FieldGroup>
-            );
-    }
-});
+    const rv= RangeValues.make( useZ ? ZSCALE : request.lowerWhich,
+        request.lowerRange,
+        useZ ? ZSCALE :request.upperWhich,
+        request.upperRange,
+        request.DR, request.BP, request.WP,
+        request.gamma, request.algorithm, request.zscaleContrast,
+        request.zscaleSamples, request.zscaleSamplesPerLine);
 
-export default ColorDialog;
+    return RangeValues.serializeRV(rv);
+}
```

Comment:
got what you made on the 'layered fieldGroup, very clear and very good for tab case. I will try to apply the way to ImageSelectPanel for 3 color plots. Thanks.